code/modules/mob/living/carbon/superior_animal/robots/subtype/mining/mining_drone.dm

```diff
@@ -0,0 +1,130 @@
+/mob/living/carbon/superior_animal/robot/mining
+	name = "Generic Mining Drone"
+	desc = "A small, unbranded drone, it has a drill and seem to search for shiny stuff."
+	gender = NEUTER
+	icon = 'icons/mob/battle_roomba.dmi'
+	icon_state = "mining_drone"
+	faction = "neutral"
+	viewRange = 6
+	move_to_delay = 3
+	randpixel = 0
+	mob_size = MOB_SMALL
+	do_gibs = TRUE
+	light_range = 5
+	light_color = COLOR_LIGHTING_CYAN_BRIGHT
+	cleaning = FALSE
+
+	// Unique vars
+	var/target // Where we want to go
+	var/wandering = TRUE // Do we wandering around?
+	var/pickup = TRUE // Do we pick up the ore?
+	var/mining = TRUE // Do we mine rocks?
+
+/mob/living/carbon/superior_animal/robot/mining/examine(mob/user)
+	..()
+	to_chat(user, SPAN_NOTICE("[src]'s screen flashes and show \his operating parameters : "))
+	to_chat(user, SPAN_NOTICE("	Wandering : [wandering ? "Active" : "Not Active"]"))
+	to_chat(user, SPAN_NOTICE("	Ore Gathering : [pickup ? "Active" : "Not Active"]"))
+	to_chat(user, SPAN_NOTICE("	Mining : [mining ? "Active" : "Not Active"]"))
+
+/mob/living/carbon/superior_animal/robot/mining/Life()
+	..()
+
+	if(!look_around() && wandering) // If we didn't find anything of value and can wandering
+		target = pick(oview(viewRange + 1, src)) // Go somewhere random
+
+	// Are we going to a minable turf despite not being supposed to?
+	if(istype(target, /turf/simulated/mineral && !mining))
+		target = null // reset the target
+
+	// Are we going to ores on the ground despite not being supposed to?
+	if(istype(target, /obj/item/ore && !pickup))
+		target = null // reset the target
+
+	if(target) // Do we have a destination?
+		walk_to(src, target, 1, move_to_delay) // Go there
+	else
+		walk_to(src, 0) // Or else stop
+
+/mob/living/carbon/superior_animal/robot/mining/death()
+	drop_loot()
+	..()
+
+/mob/living/carbon/superior_animal/robot/mining/attack_hand(mob/user as mob)
+	if(user.a_intent == I_HELP) // Are we on help intent?
+		var/possible_choices = list("Wandering", "Ore Gathering", "Ore Mining")
+		var/choice = input(user, "Which parameter do you want to toggle in the mining bot?", "Parameters", null) as null|anything in possible_choices
+		switch(choice)
+			if("Wandering")
+				wandering = !wandering
+				to_chat(user, "[src] will [wandering ? "" : "no longer"] wander.")
+			if("Ore Gathering")
+				pickup = !pickup
+				to_chat(user, "[src] will [pickup ? "" : "no longer"] take ore on the ground.")
+			if("Ore Mining")
+				mining = !mining
+				to_chat(user, "[src] will [mining ? "" : "no longer"] mine ore.")
+	else ..()
+
+/mob/living/carbon/superior_animal/robot/mining/attackby(obj/item/W as obj, mob/user as mob)
+	var/obj/item/T // Define the tool variable early on to avoid compilation problem and to allow us to use tool-unique variables
```

Comment:
Tiny ass nitpick- this variable would probably be better put inside the if statement below it, so it doesn't get created and deleted every time something touches the object.

Reply:
True, but I copy-pasted that part from the handmade bot, and IIRC back then DM threw a fit